rust/src/file.rs

```diff
@@ -153,7 +152,19 @@ impl File {
             self.header.voxel_type)?;
 
         // build Morton-order iterator
-        let iter = Iter::new(self.header.file_len_log2 as u32, dst_box_boxes)?;
+        let mut iter = Iter::new(self.header.file_len_log2 as u32, dst_box_boxes)?.peekable();
+
+        if self.header.block_type != BlockType::Raw {
```

Comment:
This is not needed for the whole-file case. Please leaves this complexity out for this pull request.